benchmark/scenarios/write_request_load.py

```diff
@@ -208,151 +77,90 @@ def _get_dataset_node(self, nodes):
+    def _set_dataset_id(self, dataset_id):
         """
-        Meassures rate and verifies that the rate haven't decreased
-        and that the scenario is not overloaded - an scenario would be
-        overloaded if there were too many outstanding requests.
-
-        :raise: `RequestRateTooLow` if the rate has dropped.
-        :raise: `RequestOverload` if the scenario is overloaded.
+        Function to set the value of the dataset id once the dataset
+        has been created
+        :param nodes: listed nodes
         """
-        rate = self.rate_measurer.rate()
-        if rate < self.request_rate:
-            self._fail(WRequestRateTooLow(rate))
-
-        if self.rate_measurer.outstanding() > self.max_outstanding:
-            self._fail(WRequestOverload())
+        self.dataset_id = dataset_id
 
-    def run_scenario(self, dataset):
+    def run_setup(self):
         """
-        :param dataset `Dataset` we will use to run the write scenario
-        :return: A `Deferred` that fires when the desired scenario is
+        Executes the setup and starts running the write scenario.
+
+        :return: A Deferred that fires when the desired scenario is
             established (e.g. that a certain load is being applied).
         """
-        # The fist thing we need to do before actually running the scenario
-        # is to update the dataset_id, as we need the information to do the
-        # write requests to generate the load
-        self.dataset_id = dataset.dataset_id
-
-        self.loop.start(interval=1)
+        # List all the nodes registered in the control service
+        d = self.control_service.list_nodes()
+        d.addCallback(self._get_dataset_node)
+        # Once we have the list of nodes, we will create a dataset.
+        # We cannot start the scenario until we have a working dataset, so
+        # `_create_dataset` will work like a setup of the write scenario
+        d.addCallback(self._create_dataset)
+        d.addCallback(self._set_dataset_id)
+        return d
 
-        def reached_target_rate():
-            return self.rate_measurer.rate() >= self.request_rate
+    def make_request(self):
+        if self.dataset_node is None:
+            # Rise exception
+            raise DatasetCreationTimeout
+        return self.control_service.move_dataset(
+            self.dataset_node.uuid,
+            self.dataset_id
+            )
 
-        def handle_timeout(failure):
-            failure.trap(CancelledError)
-            raise WRequestRateNotReached
 
-        # Loop until we reach the expected rate, or we timeout
-        waiting_for_target_rate = loop_until(self.reactor,
-                                             reached_target_rate,
-                                             repeat(1))
-        timeout(self.reactor, waiting_for_target_rate, self.timeout)
-        waiting_for_target_rate.addErrback(handle_timeout)
+@implementer(IScenario, IScenarioSetup)
+class WriteRequestLoadScenario(object):
```

Comment:
Same comment as for ReadRequest - make this a factory function returning a `RequestLoadScenario`.